### PR TITLE
[bdx] Added BlockQueryWithSkip message and BlockCounter field to BlockData

### LINKED ISSUE #8715: Add BlockQueryWithSkip to BDX messages
#### Problem
There was a new BDX message added to spec in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/3442. The codebase needs to be updated to reflect this and support this new message.

 #### Proposed Solution
Support BlockQueryWithSkip message

### LINKED ISSUE #11292: BDX: Add BlockCount field to BlockData
#### Problem
`TransferSession` emits `BlockData` structs when it receives a `Block` message from a BDX exchange. This `BlockData` struct does not currently include the `BlockCount` field, since only Synchronous BDX is implemented right now, so Blocks will always come one at a time.

Once Asynchronous BDX is implemented, `BlockData` will need to also contain the `BlockCount` field, since it is not guaranteed that messages will come in order. The application will need to order them using `BlockCount`.

 #### Proposed Solution
* add `BlockCount` to `BlockData` struct
* make sure `BlockCount` is filled in when `Block` messages are received, and also modify existing unit tests

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
* Fixes #8715
* Fixes #11292

#### Change overview
Added BlockCounter field to BlockData
Added BlockQueryWithSkip message to BDX messages

#### Testing
* Added unit tests to validate the block counter
* Added unit test to validate the BlockQueryWithSkip message
* Manually tested the BlockQueryWithSkip message with Linux OTA Requestor and Provider example.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/protocols/bdx/BdxMessages.h [hunk 1]
```diff
@@ -309,6 +310,28 @@ struct DataBlock : public BdxMessage
...
 using BlockEOF = DataBlock;
 
+/**
+ * A struct for representing BlockQueryWithSkip messages
+ */
+struct BlockQueryWithSkip : public BdxMessage
+{
+    /**
+     * @brief
+     *  Equality check method.
```
Comment (andy31415) [inline src/protocols/bdx/BdxMessages.h:320]: nit: please delete this doc-comment: it does not  seem to help me read the code (it is obvious what operator== is) and takes up 4 vertical lines.

File: src/protocols/bdx/BdxMessages.h [hunk 2]
```diff
@@ -309,6 +310,28 @@ struct DataBlock : public BdxMessage
 using Block    = DataBlock;
 using BlockEOF = DataBlock;
 
+/**
+ * A struct for representing BlockQueryWithSkip messages
```
Comment (andy31415) [inline src/protocols/bdx/BdxMessages.h:314]: Could you rephrase this comment to not re-read code? It seems obvious that `struct BlockQueryWithSkip` is a "struct for BlockQueryWithSkip"

**Review by andy31415 — APPROVED** [review]

File: src/protocols/bdx/BdxMessages.h (hunk 2 above)
Comment (shubhamdp) [reply src/protocols/bdx/BdxMessages.h]: Removed the unnecessary comments

File: src/protocols/bdx/BdxMessages.h (hunk 1 above)
Comment (shubhamdp) [reply src/protocols/bdx/BdxMessages.h]: Removed the unnecessary comment
